Move guide: Scaling svg background images (#38431)
mdn/content: files/en-us/web/css/css_backgrounds_and_borders/scaling_of_svg_backgrounds/index.md

---
title: Scaling of SVG backgrounds
slug: Web/CSS/CSS_backgrounds_and_borders/Scaling_of_SVG_backgrounds
page-type: guide
---

{{CSSRef}}

Given the flexibility of SVG images, there's a lot to keep in mind when using them as background images with the {{ cssxref("background-image") }} property, and even more to keep in mind when also scaling them using the {{ cssxref("background-size") }} property. This article describes how scaling of SVG images is handled when using these properties.

> ## The algorithm, in summary

Reviewer: Written as if the reader knows what this algorithm is, whereas they probably don't, so it comes across as mysterious. Maybe give it a name for now?

1. If {{ cssxref("background-size") }} specifies a fixed dimension (that is, percentages and relative units are fixed by their context), that dimension wins.
2. If the image has an intrinsic ratio (that is, its width:height ratio is constant, such as 16:9, 4:3, 2.39:1, 1:1, and so forth), the rendered size preserves that ratio.
3. If the image specifies a size, and the size isn't modified by `constrain` or `cover`, that specified size wins.
4. If none of the above cases are met, the image is rendered at the same size as the background area.

It's worth noting that the sizing algorithm only cares about the image's dimensions and proportions, or lack thereof. An SVG image with fixed dimensions will be treated just like a raster image of the same size.

> [!NOTE]
> If you are trying to stretch your SVG to a different {{glossary("aspect ratio")}} with CSS—for example in order to stretch it over the page background—make sure your SVG includes `preserveAspectRatio="none"`. Find out more about {{svgattr("preserveAspectRatio")}}.

## Source image examples

> Before diving in to look at the results of using different kinds of source images and seeing how they look when used with {{ cssxref("background-size") }}, it would be helpful to look at a few example source images that have different dimensions and sizing settings. These are the images we will be using for our `background-image` values in our examples. The browser renders {{SVGelement("svg")}} images as `300px` wide and `150px` tall by default.

Reviewer: ```suggestion
Before diving in to look at the results of using different kinds of SVG source images and seeing how they look when used with {{ cssxref("background-size") }}, it will be helpful to look at a few example source images that have different dimensions and sizing settings, which we will later use as our `background-image` values in our examples. The browser renders {{SVGelement("svg")}} images as `300px` wide and `150px` tall by default.
```

```css
div {
  width: 300px;
  height: 200px;
  background-repeat: no-repeat;
  border: 2px solid black;
}
```

### Specifying fixed lengths for both dimensions

If you use {{ cssxref("background-size") }} to specify fixed lengths for both dimensions, those lengths are always used, per rule 1 above. In other words, the image will always get stretched to the dimensions you specify, regardless of whether or not the source image has specified its dimensions and/or aspect ratio.

#### No dimensions or intrinsic ratio

In this example, the image has no dimensions or intrinsic ratio set:

```html hidden live-sample___scaling1
<div></div>
```

```css hidden live-sample___scaling1
div {
  width: 300px;
  height: 200px;
  background-repeat: no-repeat;
  border: 2px solid black;
}
```

```css live-sample___scaling1
div {
  background-image: url(no-dimensions-or-ratio.svg);
  background-size: 125px 175px;
}
```

> {{ EmbedLiveSample('scaling1', 200, 230) }}

Reviewer: None of these examples seem to be rendering properly in the preview. You can see the border set on the div, but no SVG image. Do they render in your local version?

Author: yes. It took 6 hours to get the page to render correctly, but it did start rendering correctly... finally
